### PR TITLE
Convert project to Gradle+Kotlin

### PR DESCRIPTION
**Pull request by jmfayard**
While I was doing this small pull request earlier, I realized that your project was exactly the playground I needed (neither trivial not too large, with tests) for something I had in mind : 

Showing by example how to convert a Gradle project to the Kotlin DSL. 

It will be a follow-up to this article I wrote and I plan to show it at a Kotlin Meetup in Berlin. 

https://blog.kotlin-academy.com/gradle-kotlin-the-missing-piece-of-the-puzzle-7528a85f0d2c

If you prefer to stay with Gradle/Groovy, no hard feeling :)

It should work fine with `Intellij IDEA 2018.+` and `Android Studio 3.2+`,  tell me if you encounter problems.

### CODE REVIEW

**Comment by jmfayard** [comment]
One module can have either have one.
But we can have half of the modules in groovy, half in kotlin
We can also have the kotlin build and have the groovy version with a different name like `build.gradle.groovy` so that it's not part of the build

**Comment by rozkminiacz** [comment]
Yeah, that's a good solution. Project built fully in Kotlin and Groovy files with .Groovy sufix excluded from build. It also requires proper explanation in readme.

**Comment by jmfayard** [comment]
@rozkminiacz Done[1] 

I even added a script `./toggle-kotlin-groovy.sh`  to switch between the two builds

[1] Github has some problems today so it may not appear so

**Comment by rozkminiacz** [comment]
Looks good!
Nice work with toggle script. Merging.

File: buildSrc/src/main/kotlin/Versions.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * Run again
+ *   `$ ./gradlew syncLibs`
+ * to update this file */
+object Versions {
+    const val jmfayard_github_io_gradle_kotlin_dsl_libs_gradle_plugin: String =
+            "0.2.3" // available: milestone=0.2.4
+
+    const val junit: String = "4.12" // up-to-date
+
+    const val junit_jupiter_api: String = "5.3.1" // up-to-date
```
[the remark this answers is not in the document]
Comment (jmfayard) [reply buildSrc/src/main/kotlin/Versions.kt]: sorry, I forgot to reply. I follow the Kotlin style guide by default but in this case, I decided to that it was more important to be closer to what the maven coordinate is.

**Pull request merged**